src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CSharp.Analyzers/AllowHtmlAttributeAnalyzer.cs

```diff
@@ -22,15 +22,16 @@ public class AllowHtmlAttributeAnalyzer : DiagnosticAnalyzer
 
         public override void Initialize(AnalysisContext context)
         {
-            if (context is null)
-            {
-                throw new ArgumentNullException(nameof(context));
-            }
-
             context.EnableConcurrentExecution();
             context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze);
-
-            context.RegisterSyntaxNodeAction(AnalyzeAttribute, SyntaxKind.Attribute);
+            context.RegisterCompilationStartAction(context =>
+            {
+                var allowHtmlType = context.Compilation.GetTypeByMetadataName(AllowHtmlAttributeName);
```

Comment:
From my understanding, the analyzer can only report a diagnostic if this type exists. So I'm checking that first before registering callbacks.

This should improve the performance of this analyzer by preventing unneeded callbacks to be made.

It's also worth trying to update the analyzer to not rely on syntax in a follow-up. Which is both more performant and more reliable.

Reply:
The Upgrade Assistant runs source updaters towards the end of the upgrade process. By the time this is run, the System.Web.* references will have already been removed so we probably won't be able to resolve the type to `System.Web.Mvc.AllowHtmlAttribute`. Instead, we have to look for syntax that probably refers to that type (because the simple name matches and it doesn't resolve to any other type).